airlock/forms.py

```diff
@@ -11,6 +11,15 @@ class MultiValueField(forms.Field):
     widget = forms.SelectMultiple
 
 
+class InternalRedirectField(forms.CharField):
+    """Ensure is internal url path, not absolute url."""
+
+    def validate(self, value):
+        super().validate(value)
+        if value[0] != "/":
+            raise forms.ValidationError("Must be absolute url path")
```

Comment:
Should the error message be "Must be relative url path"? Or "Must not be absolute ..."?

Reply:
So, there's two possible meanings of absolute here. It must start with a /, i.e not a relative path, and it must *not* have a full domain

I did this after the security scanner pointed out that next_url could be spoofed, and used to redirect somewhere else.  I don't think it's a real concern for airlock atm, but it seemed sensitive to restrict the redirect to a local path, not an external site.

Reply:
"absolute url path" to me meant _including_ the full domain. I don't disagree with what's it's doing, I just found the error message confusing. I guess this is a hidden field and the error shouldn't be shown to users, so it's not too important.

Reply:
Right, the word "path" is load bearing here - there's a difference between an "absolute url" and an "absolute url path". This is the terminology I am using: https://stackoverflow.com/questions/904046/absolute-urls-relative-urls-and

The phrase "Must be relative url path" is not right IMO.  next_url must be a relative url, but an absolute url path - so perhaps we put that?

But your point about not being shown to the user has me thinking.  Maybe rather than a form error, we just 403? Then there's no wording to worry about?